apps/obsidian/src/components/DiscourseContextView.tsx

```diff
@@ -0,0 +1,144 @@
+import { ItemView, TFile, WorkspaceLeaf } from "obsidian";
+import { createRoot, Root } from "react-dom/client";
+import DiscourseGraphPlugin from "../index";
+import { getDiscourseNodeFormatExpression } from "../utils/getDiscourseNodeFormatExpression";
+
+export const VIEW_TYPE_DISCOURSE_CONTEXT = "discourse-context-view";
+
+interface DiscourseContextProps {
+  activeFile: TFile | null;
+  plugin: DiscourseGraphPlugin;
+}
+
+function DiscourseContext({ activeFile, plugin }: DiscourseContextProps) {
+  const extractContentFromTitle = (
+    format: string | undefined,
+    title: string,
+  ): string => {
+    if (!format) return "";
+    const regex = getDiscourseNodeFormatExpression(format);
+    const match = title.match(regex);
+    return match?.[1] ?? title;
+  };
+
+  const renderContent = () => {
+    if (!activeFile) {
+      return <div>No file is open</div>;
+    }
+
+    const cache = plugin.app.metadataCache.getFileCache(activeFile);
```

Comment:
The name `cache` trips me up here.  It is slightly ambiguous and makes me thing we are managing a cache somewhere for this.  What about something like `fileMetadata`, `metadata`, or `data`.